File: Aplicación Escritorio/ModeloTDO/ValoracionDevueltaDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ModeloDTO
{
    public class ValoracionDevueltaDTO
    {
        public int idValoracion { get; set; }
        public int dificultad { get; set; }
        public int belleza { get; set; }
        public int interesCultural { get; set; }
        public String nomUsuario { get; set; }
        public String nomRuta { get; set; }

        public DateTimeOffset fecha { get; set; }

    }
}

File: Springboot_API/RetaCantabria/src/main/java/org/example/DTO/ValoracionDevueltaDTO.java
package org.example.DTO;

import org.example.Entidades.Valoracion;

import java.time.OffsetDateTime;
import java.time.ZoneOffset;

public record ValoracionDevueltaDTO (
        int idValoracion,
        int dificultad,
        int belleza,
        int interesCultura,
        String nomUsuario,
        String nomRuta,
        OffsetDateTime fecha
){
    public ValoracionDevueltaDTO(Valoracion val) {
        this(
                val.getId(),
                val.getDificultad(),
                val.getBelleza(),
                val.getInteresCultural(),
                val.getUsuario().getNombre(),
                val.getRuta().getNombre(),
                OffsetDateTime.of(val.getFecha(), ZoneOffset.of("-08:00"))
        );
    }
}
